src/library/scala/collection/convert/JavaCollectionWrappers.scala

```diff
@@ -56,7 +57,15 @@ private[collection] object JavaCollectionWrappers extends Serializable {
   }
 
   @SerialVersionUID(3L)
-  class IterableWrapper[A](val underlying: Iterable[A]) extends ju.AbstractCollection[A] with IterableWrapperTrait[A] with Serializable
+  class IterableWrapper[A](val underlying: Iterable[A]) extends ju.AbstractCollection[A] with IterableWrapperTrait[A] with Serializable {
+    import scala.runtime.Statics._
+    override def equals(other: Any): Boolean =
+      other match {
+        case other: IterableWrapper[A @uc] => underlying.equals(other.underlying)
```

Comment:
That's not a safe `@unchecked`. Consider making `A` covariant and then match on `IterableWrapper[Any]`.

Reply:
Thanks, cut/paste. I considered just `[_]`.